### PR TITLE
Applicant view

### PR DESCRIPTION
**Pull request by smwatts**
A bit different than what we talked about but here the AC has the ability to view all students will be re-directed to a separate page to edit/disqualify (worked with @j2kan to determine the best implementation here)

screen shot 2018-03-03 at 8 49 59 am

The display/export follows the current format Smucker uses:
screen shot 2018-03-03 at 8 50 07 am

If no apps exist:
screen shot 2018-03-01 at 3 43 35 pm

Also made a few cosmetic changes for example:
screen shot 2018-03-01 at 3 45 06 pm

Dare you to find a bug!

### CODE REVIEW

**Comment by j2kan** [comment]
@smwatts missing:
Documentation for instructions to test and system documentation.

**Review by amyleblond — APPROVED** [review]
All worked sarah!

File: Documentation/README.md [hunk 1]
```diff
@@ -9,21 +9,53 @@
 3.
 
 ### System Instructions:
-0. Login to the Application - (https://team4.uwaterloo.ca/login/)
-1. 
-2.
-3. [Modify Apps](404)
+1. [Review Applicants]
+	* Function: This page allows the AC to view basic student information and access links to edit student information, disqualify students from the TA process and change student rankings for courses. This page also allows the AC to export all current student informaiton.
```
Comment (j2kan) [inline Documentation/README.md]: information.

File: Documentation/README.md [hunk 2]
```diff
@@ -9,21 +9,53 @@
...
+	* Scenarios for Use: 
+		** 1 - The AC can export a complete list of applicant information to send to the GC. 
+		** 2 - The AC can access linked to edit student information, disqualify students from the TA process and change student rankings for courses.
```
Comment (j2kan) [inline Documentation/README.md]: reword this sentence.

File: Documentation/README.md [hunk 3]
```diff
@@ -9,21 +9,53 @@
 3.
 
 ### System Instructions:
-0. Login to the Application - (https://team4.uwaterloo.ca/login/)
-1. 
-2.
-3. [Modify Apps](404)
+1. [Review Applicants]
```
Comment (j2kan) [inline Documentation/README.md]: add back step 0. login to application

File: Documentation/README.md [hunk 4]
```diff
@@ -9,21 +9,53 @@
...
+	* Function: This page allows the AC to view basic student information and access links to edit student information, disqualify students from the TA process and change student rankings for courses. This page also allows the AC to export all current student informaiton.
+	* Scenarios for Use: 
+		** 1 - The AC can export a complete list of applicant information to send to the GC. 
+		** 2 - The AC can access linked to edit student information, disqualify students from the TA process and change student rankings for courses.
+	* How to Use: 
+		* Login to the Application - (https://team4.uwaterloo.ca/login/)
+		* Open Step 5, Review Applicants - (https://team4.uwaterloo.ca/taform/applicants.html)
+		* Review the students that have applied.
+		* Select 'Edit student info' on the student you would like to modify course preference for.
```
Comment (j2kan) [inline Documentation/README.md]: false statement.

Comment (j2kan) [reply Documentation/README.md]: modify student's personal information.

Comment (j2kan) [reply Documentation/README.md]: oh it's backwards. you mean edit course rankings

File: Documentation/README.md [hunk 5]
```diff
@@ -9,21 +9,53 @@
...
+		* Select 'Edit student info' on the student you would like disqualify or edit the basic information of.
+		* Export the student information listed in the table using the "export applicants" button in the top right.
+	* Template: applicants.html
```
Comment (j2kan) [inline Documentation/README.md]: remove line 24-26

File: Documentation/README.md [hunk 6]
```diff
@@ -9,21 +9,53 @@
...
+	* Views: def applicants(request)
+	* Models: class Student(models.Model)
+2. [Edit Student Information]
+	* Function: This page allows the AC to edit basic student information and disqualify students from the matching process. 
+	* Scenarios for Use: 
+		** 1 - A student has made a mistake on their application form and would like the AC to change it. 
+		** 2 - The AC want to disqualify a student from the process or prevent a duplicate application from moving forward.
+	* How to Use: 
+		* Login to the Application - (https://team4.uwaterloo.ca/login/)
```
Comment (j2kan) [inline Documentation/README.md]: remove. refer to step 0

File: Documentation/README.md [hunk 7]
```diff
@@ -9,21 +9,53 @@
...
+		* Open Step 5, Review Applicants - (https://team4.uwaterloo.ca/taform/applicants.html)
+		* Review the students that have applied.
+		* Select 'Edit student info' on the student you would like disqualify or edit the basic information of.
+		* The screen you will see lists the students basic information for the upcoming term as well as the ability to "disqualify" the applicant.
+		* You are able to change the students basic informaiton at this time.
+		* If you decide to "disqualify" an applicant, they will not appear in the instructor rankings or the final matches.
+		* When you select submit you will be re-directed to the Applicant page where you can see the changes you have made for that student.
+	* Template: student_form.html
```
Comment (j2kan) [inline Documentation/README.md]: remove these too

File: Documentation/README.md [hunk 8]
```diff
@@ -9,21 +9,53 @@
...
+	* Models: class StudentForm(ModelForm)
+3. [Modify Apps]
 	* Function: This page allows the AC to modify any students preference for teaching all courses offered in the upcoming term. 
-	* Scenarios for Use: 1 - A student has made a mistake on their application form and would like the AC to change it. 2 - The AC needs to force a match between a student and class.
+	* Scenarios for Use: 
+		** 1 - A student has made a mistake on their application form and would like the AC to change it. 
+		** 2 - The AC needs to force a match between a student and class.
 	* How to Use: 
-		* Open Step X, Modify or Delete Student Applications - ()
+		* Login to the Application - (https://team4.uwaterloo.ca/login/)
```
Comment (j2kan) [inline Documentation/README.md]: remove.

File: Documentation/README.md [hunk 9]
```diff
@@ -9,21 +9,53 @@
...
+		* Open Step 5, Review Applicants - (https://team4.uwaterloo.ca/taform/applicants.html)
 		* Review the students that have applied.
-		* Select 'Make Preference Changes' on the student you would like to modify course preference for.
-		* The screen you know see lists all of the courses for the upcoming term, along with the student's preference to TA each of them.
+		* Select 'Edit rankings' on the student you would like to modify course preference for.
+		* The screen you will see lists all of the courses for the upcoming term for the selected student. You will see the student's preference to TA each course.
```
Comment (j2kan) [inline Documentation/README.md]: This screen.

Comment (j2kan) [reply Documentation/README.md]: student's preference to TA `for` each course.

File: taworks/README.md [hunk 10]
```diff
@@ -109,18 +109,23 @@ _after running the queries below, you will need to press `q` to get out of it_
 13. [Go on the algorithm page](https://team4.uwaterloo.ca/taform/algorithm.html)
     * Download both csv exports from the [export page](https://team4.uwaterloo.ca/taform/export.html)
         * Use these csv exports as a point of reference for what is saved by the system
-    * Run the algorithm. Verify it won't work without applicants or courses in the database.
```
Comment (j2kan) [inline taworks/README.md]: add these instructions back, there is no longer a Go! button.

File: taworks/taform/templates/taform/home.html [hunk 11]
```diff
@@ -42,38 +42,46 @@ <h1><legend>TA Application Home Page</legend></h1>
...
-            <td class="tg-yw4l"><a href="application.html">TA Application</a></td>
-            <td class="tg-yw4l">When the upcoming terms courses have been added
-             to the system, the AC must make the TA Application Form 'live'. 
-             This allows graduate students to apply for a TA position. The AC must 
-            then close the form when the application period has ended. 
-            The AC may also preview the form here.</td>
+            <td class="tg-yw4l">5</td>
+            <td class="tg-yw4l"><a href="applicants.html">Review Applicants</a></td>
+            <td class="tg-yw4l">The AC is responsible for ensuring all applicants are
+            eligible to hold TA positions. The AC can export applicant informaiton to
```
Comment (j2kan) [inline taworks/taform/templates/taform/home.html]: to hold a TA position.

Comment (j2kan) [reply taworks/taform/templates/taform/home.html]: spelt information

File: taworks/taform/views.py [hunk 12]
```diff
@@ -25,6 +25,19 @@
 from django.db.models import Count, Case, When, IntegerField, Avg
 import pulp
 import math
+from django.contrib import messages
```
Comment (j2kan) [inline taworks/taform/views.py]: remove, not using.

File: taworks/taform/views.py [hunk 13]
```diff
@@ -259,12 +306,13 @@ def algorithm(request):
...
     AC = authenticated(request)
     df = pd.DataFrame(list(models.Assignment.objects.all().values()))
+    df_elg_students = pd.DataFrame(list(models.Student.objects.all().filter(is_disqualified = False).values()))
     max_date = None
     matches =  pd.DataFrame()
     courses_supply = pd.DataFrame()
     students_supply = pd.DataFrame()
     context = None
-    if not df.empty:
+    if not df.empty and not df_elg_students.empty:
```
Comment (j2kan) [inline taworks/taform/views.py]: good opportunity to rename df to df_assignments

**Review by j2kan — APPROVED** [review]
mostly wording stuff.
logic is great, you put in some good work!

lgtm

**Review by j2kan — DISMISSED** [review]
ps fix all before merge!

File: Documentation/README.md (hunk 3 above)
Comment (smwatts) [reply Documentation/README.md]: done

File: Documentation/README.md (hunk 1 above)
Comment (smwatts) [reply Documentation/README.md]: done

File: Documentation/README.md (hunk 2 above)
Comment (smwatts) [reply Documentation/README.md]: done

File: Documentation/README.md (hunk 4 above)
Comment (smwatts) [reply Documentation/README.md]: done

File: Documentation/README.md (hunk 5 above)
Comment (smwatts) [reply Documentation/README.md]: done

File: Documentation/README.md (hunk 6 above)
Comment (smwatts) [reply Documentation/README.md]: done

File: Documentation/README.md (hunk 7 above)
Comment (smwatts) [reply Documentation/README.md]: done

File: Documentation/README.md (hunk 8 above)
Comment (smwatts) [reply Documentation/README.md]: done

File: Documentation/README.md (hunk 9 above)
Comment (smwatts) [reply Documentation/README.md]: done

File: taworks/README.md (hunk 10 above)
Comment (smwatts) [reply taworks/README.md]: done

File: taworks/taform/templates/taform/home.html (hunk 11 above)
Comment (smwatts) [reply taworks/taform/templates/taform/home.html]: done

File: taworks/taform/views.py (hunk 12 above)
Comment (smwatts) [reply taworks/taform/views.py]: done

File: taworks/taform/views.py (hunk 13 above)
Comment (smwatts) [reply taworks/taform/views.py]: it is

**Pull request merged**